File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DocumentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    knowledge_base_id: int
    file_name: str
    original_name: str
    file_type: str
    storage_path: str
    parse_status: str
    index_status: str
    chunk_count: int
    error_message: str | None
    uploaded_at: datetime
    updated_at: datetime



class DocumentUploadResponse(BaseModel):
    document_id: int
    status: str
    file_name: str


class DocumentReindexResponse(BaseModel):
    document_id: int
    status: str

File: frontend/src/types/document.ts
export type DocumentRecord = {
  id: number;
  knowledge_base_id: number;
  file_name: string;
  original_name: string;
  file_type: string;
  storage_path: string;
  parse_status: string;
  index_status: string;
  chunk_count: number;
  error_message: string | null;
  uploaded_at: string;
  updated_at: string;
};

export type UploadDocumentResponse = {
  document_id: number;
  status: string;
  file_name: string;
};

export type ReindexDocumentResponse = {
  document_id: number;
  status: string;
};
